### PR TITLE
migrate 17 enzyme tests to testing library MAASENG-1710

### PR DESCRIPTION
**Pull request by petermakowski**
## Done

- migrate the following tests from enzyme to @testing-library/react:
  - NodeTestDetailsLogs
  - LXDPowerFields
  - SwitchField
  - TableActionsDropdown
  - TableHeader
  - TagField
  - TagLinks
  - FagLinks
  - NotFound
  - LxdTable
  - LXDClusterSettings
  - UpdateCertificateFields
  - DangerZoneCard
  - ReleaseFormFields
  - MachinePCIDevices
  - AddHardwareMenu
  - MachineTestStatus

### MAAS deployment

To run this branch you will need access to one of the following MAAS deployments:

- [Karura](/HACKING.md#karura)
- [Bolla](/HACKING.md#bolla)
- [A development MAAS](/HACKING.md#development-deployment)
- [A local snap MAAS](/HACKING.md#snap-deployment) (this will not usually have machines)

### Running the branch

You can run this branch by:

- Serving with [dotrun](/HACKING.md#maas-ui-development-setup)
- [Building in a development MAAS](/HACKING.md#running-maas-ui-from-a-development-maas)

## Fixes

Fixes: https://warthogs.atlassian.net/browse/MAASENG-1710

## Launchpad issue

Related Launchpad maas issue in the form `lp#number`.

## Backports

In general, please propose fixes against _main_ rather than release branches (e.g. 2.7), unless the fix is only applicable for that specific release. Please apply backport labels to the PR (e.g. "Backport 2.7") for the appropriate releases to target.

Only bug and security fixes should be backported, new features should only land in main.

## Screenshots

It could be helpful to provide some screenshots to aid in QAing the change.

### CODE REVIEW

File: src/app/base/components/CertificateFields/CertificateFields.tsx [hunk 1]
```diff
@@ -27,6 +27,7 @@ export const CertificateFields = ({
       <p>Certificate</p>
       <Input
         checked={shouldGenerateCert}
+        data-testid="generate-certificate"
```
Comment (ndv99) [inline src/app/base/components/CertificateFields/CertificateFields.tsx:30]: Does this need a test-id to work or can we use `getByRole`?

File: src/app/base/components/PowerTypeFields/LXDPowerFields/LXDPowerFields.test.tsx [hunk 2]
```diff
@@ -21,9 +21,9 @@ describe("LXDPowerFields", () => {
   });
 
   it("can be given a custom power parameters name", () => {
-    const field = powerFieldFactory({ name: "field" });
+    const field = powerFieldFactory({ name: "field", label: "custom field" });
     const store = mockStore(state);
-    const wrapper = mount(
+    render(
```
Comment (ndv99) [inline src/app/base/components/PowerTypeFields/LXDPowerFields/LXDPowerFields.test.tsx:26]: This and the other tests in this file should be able to use `renderWithMockStore`

File: src/app/base/components/TableHeader/TableHeader.test.tsx [hunk 3]
```diff
@@ -1,22 +1,24 @@
...
 import { SortDirection } from "app/base/types";
+import { render, screen } from "testing/utils";
 
 describe("TableHeader ", () => {
   it("renders a div if no onClick prop is present", () => {
-    const wrapper = shallow(<TableHeader>Text</TableHeader>);
-    expect(wrapper.find("Button").exists()).toBe(false);
-    expect(wrapper.find("div").exists()).toBe(true);
+    const { container } = render(<TableHeader>Text</TableHeader>);
+    expect(screen.queryByRole("button")).toBeNull();
```
Comment (ndv99) [inline src/app/base/components/TableHeader/TableHeader.test.tsx:12]:
```suggestion
    expect(screen.queryByRole("button")).not.toBeInTheDocument();
```

File: src/app/base/views/NotFound/NotFound.test.tsx [hunk 4]
```diff
@@ -1,43 +1,22 @@
...
-        <MemoryRouter initialEntries={[{ pathname: "/404", key: "testKey" }]}>
-          <NotFound includeSection />
-        </MemoryRouter>
-      </Provider>
-    );
-    expect(wrapper.find("MainContentSection").exists()).toBe(true);
+    const { container } = renderWithBrowserRouter(<NotFound includeSection />, {
+      route: "/404",
+    });
+    expect(container.querySelector("section")).toBeInTheDocument();
```
Comment (ndv99) [inline src/app/base/views/NotFound/NotFound.test.tsx:20]: Could be worth also checking the contents of the section to make sure it's correct

File: src/app/kvm/views/LXDSingleDetails/LXDSingleSettings/DangerZoneCard/DangerZoneCard.test.tsx [hunk 5]
```diff
@@ -26,13 +25,13 @@ describe("DangerZoneCard", () => {
...
-    const wrapper = mount(
+    render(
       <DangerZoneCard
         hostId={1}
         message={<span data-testid="message">Delete KVM</span>}
         setSidePanelContent={setSidePanelContent}
       />
     );
-    expect(wrapper.find("[data-testid='message']").exists()).toBe(true);
+    expect(screen.getByTestId("message")).toBeInTheDocument();
```
Comment (ndv99) [inline src/app/kvm/views/LXDSingleDetails/LXDSingleSettings/DangerZoneCard/DangerZoneCard.test.tsx:35]:
```suggestion
    expect(screen.getByTestId("message")).toHaveTextContent("Delete KVM");
```

File: src/app/machines/views/MachineDetails/MachinePCIDevices/MachinePCIDevices.test.tsx [hunk 6]
```diff
@@ -27,7 +27,7 @@ describe("MachinePCIDevices", () => {
       }),
     });
     const store = mockStore(state);
-    mount(
+    render(
```
Comment (ndv99) [inline src/app/machines/views/MachineDetails/MachinePCIDevices/MachinePCIDevices.test.tsx:30]: You can replace this with the `renderWithBrowserRouter` util

**Review by ndv99 — APPROVED** [review]
Looks mostly good, a few suggestions below

File: src/app/base/views/NotFound/NotFound.test.tsx (hunk 4 above)
Comment (petermakowski) [reply src/app/base/views/NotFound/NotFound.test.tsx:20]: I agree, but this is arguably out of scope of simple migration.

**Pull request merged**